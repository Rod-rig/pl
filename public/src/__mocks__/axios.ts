import config from '../config/config';
import {matchListMock, scheduleMock, tableMock, tournamentListMock} from './';

export default {
  get: jest.fn((url) => {
    const rootUrl = `${config.apiUrl}/en`;
    const urlWithoutKey = url.split('?')[0];
    if (urlWithoutKey === `${rootUrl}/tournaments.json`) {
      return Promise.resolve({data: tournamentListMock});
    } else if (urlWithoutKey === `${rootUrl}/tournaments/test/standings.json`) {
      return Promise.resolve({data: tableMock});
    } else if (urlWithoutKey === `${rootUrl}/tournaments/test_id/results.json`) {
      return Promise.resolve({data: matchListMock});
    } else if (urlWithoutKey === `${rootUrl}/schedules/2018-07-07/schedule.json`) {
      return Promise.resolve({data: scheduleMock});
    } else {
      return Promise.reject('No such url');
    }
  }),
};
