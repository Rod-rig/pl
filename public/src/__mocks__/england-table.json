[
  {
    "position": 1,
    "w": 27,
    "d": 3,
    "l": 1,
    "goals for": 88,
    "goals against": 21,
    "points": 84,
    "teamName": "Manchester City"
  },
  {
    "position": 2,
    "w": 21,
    "d": 5,
    "l": 5,
    "goals for": 60,
    "goals against": 23,
    "points": 68,
    "teamName": "Manchester United"
  },
  {
    "position": 3,
    "w": 19,
    "d": 9,
    "l": 4,
    "goals for": 75,
    "goals against": 35,
    "points": 66,
    "teamName": "Liverpool"
  }
]