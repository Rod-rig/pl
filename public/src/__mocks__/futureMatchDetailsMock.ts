/* tslint:disable: object-literal-key-quotes object-literal-sort-keys*/
export const futureMatchDetailsMock = {
  generated_at: "2019-08-04T16:21:35+00:00",
  schema:
    "http://schemas.sportradar.com/bsa/soccer/v1/json/endpoints/soccer/match_summary.json",
  sport_event: {
    id: "sr:match:18427304",
    scheduled: "2019-08-09T19:00:00+00:00",
    start_time_tbd: false,
    tournament_round: { type: "group", number: 1, phase: "regular season" },
    season: {
      id: "sr:season:66441",
      name: "Premier League 19/20",
      start_date: "2019-08-09",
      end_date: "2020-05-18",
      year: "19/20",
      tournament_id: "sr:tournament:17",
    },
    tournament: {
      id: "sr:tournament:17",
      name: "Premier League",
      sport: { id: "sr:sport:1", name: "Soccer" },
      category: { id: "sr:category:1", name: "England", country_code: "ENG" },
    },
    competitors: [
      {
        id: "sr:competitor:44",
        name: "Liverpool FC",
        country: "England",
        country_code: "ENG",
        abbreviation: "LFC",
        qualifier: "home",
      },
      {
        id: "sr:competitor:263",
        name: "Norwich City",
        country: "England",
        country_code: "ENG",
        abbreviation: "NOR",
        qualifier: "away",
      },
    ],
    venue: {
      id: "sr:venue:579",
      name: "Anfield",
      capacity: 54074,
      city_name: "Liverpool",
      country_name: "England",
      map_coordinates: "53.430622,-2.960919",
      country_code: "ENG",
    },
  },
  sport_event_conditions: {
    venue: {
      id: "sr:venue:579",
      name: "Anfield",
      capacity: 54074,
      city_name: "Liverpool",
      country_name: "England",
      map_coordinates: "53.430622,-2.960919",
      country_code: "ENG",
    },
  },
  sport_event_status: { status: "not_started" },
};
