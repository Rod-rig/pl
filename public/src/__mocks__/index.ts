export * from "./futureMatchDetailsMock";
export * from "./matchDetailsMock";
export * from "./matchListMock";
export * from "./scheduleByDateMock";
export * from "./scheduleByTournamentId";
export * from "./tableMock";
export * from "./tournamentListMock";
export * from "./userMock";
export * from "./predictions";
