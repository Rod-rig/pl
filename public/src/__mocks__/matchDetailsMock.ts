/* tslint:disable: object-literal-key-quotes object-literal-sort-keys*/
export const matchDetailsMock = {
  generated_at: "2019-07-27T09:57:47+00:00",
  schema:
    "http://schemas.sportradar.com/bsa/soccer/v1/json/endpoints/soccer/match_summary.json",
  sport_event: {
    id: "sr:match:14735957",
    scheduled: "2018-08-10T19:00:00+00:00",
    start_time_tbd: false,
    tournament_round: {
      type: "group",
      number: 1,
      phase: "regular season",
    },
    season: {
      id: "sr:season:54571",
      name: "Premier League 18/19",
      start_date: "2018-08-10",
      end_date: "2019-05-13",
      year: "18/19",
      tournament_id: "sr:tournament:17",
    },
    tournament: {
      id: "sr:tournament:17",
      name: "Premier League",
      sport: {
        id: "sr:sport:1",
        name: "Soccer",
      },
      category: {
        id: "sr:category:1",
        name: "England",
        country_code: "ENG",
      },
    },
    competitors: [
      {
        id: "sr:competitor:35",
        name: "Manchester United",
        country: "England",
        country_code: "ENG",
        abbreviation: "MNU",
        qualifier: "home",
      },
      {
        id: "sr:competitor:31",
        name: "Leicester City",
        country: "England",
        country_code: "ENG",
        abbreviation: "LEI",
        qualifier: "away",
      },
    ],
    venue: {
      id: "sr:venue:9",
      name: "Old Trafford",
      capacity: 75635,
      city_name: "Manchester",
      country_name: "England",
      map_coordinates: "53.463150,-2.291444",
      country_code: "ENG",
    },
  },
  sport_event_conditions: {
    referee: {
      id: "sr:referee:52261",
      name: "Marriner, Andre",
      nationality: "England",
      country_code: "ENG",
    },
    referee_assistants: [
      {
        type: "first_assistant_referee",
        id: "sr:referee:79890",
        name: "Ledger, Scott",
        nationality: "England",
        country_code: "ENG",
      },
      {
        type: "second_assistant_referee",
        id: "sr:referee:88710",
        name: "Long, Simon",
        nationality: "England",
        country_code: "ENG",
      },
    ],
    venue: {
      id: "sr:venue:9",
      name: "Old Trafford",
      capacity: 75635,
      city_name: "Manchester",
      country_name: "England",
      map_coordinates: "53.463150,-2.291444",
      country_code: "ENG",
    },
    attendance: 74439,
    weather_info: {
      pitch: "good",
      weather_conditions: "good",
    },
  },
  sport_event_status: {
    status: "closed",
    match_status: "ended",
    home_score: 2,
    away_score: 1,
    winner_id: "sr:competitor:35",
    period_scores: [
      {
        home_score: 1,
        away_score: 0,
        type: "regular_period",
        number: 1,
      },
      {
        home_score: 1,
        away_score: 1,
        type: "regular_period",
        number: 2,
      },
    ],
  },
  statistics: {
    teams: [
      {
        id: "sr:competitor:35",
        name: "Manchester United",
        abbreviation: "MNU",
        qualifier: "home",
        statistics: {
          ball_possession: 46,
          shots_blocked: 1,
          shots_on_target: 6,
          throw_ins: 19,
          fouls: 11,
          goal_kicks: 7,
          shots_saved: 3,
          free_kicks: 9,
          offsides: 4,
          corner_kicks: 2,
          shots_off_target: 1,
          injuries: 1,
          yellow_cards: 2,
        },
        players: [
          {
            id: "sr:player:69378",
            name: "De Gea, David",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:19352",
            name: "Darmian, Matteo",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:606346",
            name: "Bailly, Eric",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:143334",
            name: "Lindelof, Victor",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:190839",
            name: "Shaw, Luke",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 1,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:243623",
            name: "Fred",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 1,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:285949",
            name: "Pereira, Andreas",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 1,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:111802",
            name: "Pogba, Paul",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 1,
            penalty_goals_scored: 1,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:34120",
            name: "Sanchez, Alexis",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 1,
          },
          {
            id: "sr:player:903830",
            name: "Rashford, Marcus",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:28367",
            name: "Mata, Juan",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 1,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:78893",
            name: "Lukaku, Romelu",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:908372",
            name: "McTominay, Scott",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:21764",
            name: "Fellaini, Marouane",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
        ],
      },
      {
        id: "sr:competitor:31",
        name: "Leicester City",
        abbreviation: "LEI",
        qualifier: "away",
        statistics: {
          ball_possession: 54,
          free_kicks: 15,
          throw_ins: 20,
          shots_on_target: 4,
          offsides: 2,
          fouls: 7,
          goal_kicks: 4,
          shots_blocked: 6,
          corner_kicks: 5,
          shots_saved: 4,
          shots_off_target: 4,
          yellow_cards: 1,
        },
        players: [
          {
            id: "sr:player:1076",
            name: "Schmeichel, Kasper",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:308526",
            name: "Amartey, Daniel",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 1,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:20375",
            name: "Morgan, Wes",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:149380",
            name: "Maguire, Harry",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:754594",
            name: "Chilwell, Benjamin",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:221200",
            name: "Pereira, Ricardo",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:795266",
            name: "Ndidi, Onyinye Wilfred",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:30127",
            name: "Silva, Adrien",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:356398",
            name: "Maddison, James",
            substituted_in: 0,
            substituted_out: 1,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:355152",
            name: "Gray, Demarai",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:359642",
            name: "Iheanacho, Kelechi",
            substituted_in: 0,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:123228",
            name: "Ghezzal, Rachid",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:173827",
            name: "Vardy, Jamie",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 1,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
          {
            id: "sr:player:32690",
            name: "Iborra, Vicente",
            substituted_in: 1,
            substituted_out: 0,
            goals_scored: 0,
            assists: 0,
            own_goals: 0,
            yellow_cards: 0,
            yellow_red_cards: 0,
            red_cards: 0,
          },
        ],
      },
    ],
  },
};
