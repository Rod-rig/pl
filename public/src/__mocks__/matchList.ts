export default {
  matchList: [
    {
      awayScore: 1,
      awayTeam: 'Chelsea',
      date: '28.04. 19:30',
      homeScore: 0,
      homeTeam: 'Swansea City',
      round: 36,
    },
    {
      awayScore: 0,
      awayTeam: 'Brighton and Hove Albion',
      date: '28.04. 17:00',
      homeScore: 0,
      homeTeam: 'Burnley',
      round: 36,
    },
    {
      awayScore: 0,
      awayTeam: 'Leicester City',
      date: '28.04. 17:00',
      homeScore: 5,
      homeTeam: 'Crystal Palace',
      round: 36,
    },
  ],
};
