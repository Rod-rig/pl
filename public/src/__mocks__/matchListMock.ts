/* tslint:disable: object-literal-key-quotes object-literal-sort-keys*/
export const matchListMock = {
  'generated_at': '2018-06-16T14:29:26+00:00',
  'schema': 'http://schemas.sportradar.com/bsa/soccer/v1/json/endpoints/soccer/tournament_results.json',
  'tournament': {
    'id': 'sr:tournament:16',
    'name': 'World Cup',
    'sport': {
      'id': 'sr:sport:1',
      'name': 'Soccer',
    },
    'category': {
      'id': 'sr:category:4',
      'name': 'International',
    },
  },
  'results': [
    {
      'sport_event': {
        'id': 'sr:match:13245886',
        'scheduled': '2018-06-14T15:00:00+00:00',
        'start_time_tbd': false,
        'tournament_round': {
          'type': 'group',
          'number': 1,
          'group': 'A',
          'tournament_match_number': 1,
        },
        'season': {
          'id': 'sr:season:48238',
          'name': 'World Cup 2018',
          'start_date': '2018-06-14',
          'end_date': '2018-07-16',
          'year': '2018',
          'tournament_id': 'sr:tournament:16',
        },
        'tournament': {
          'id': 'sr:tournament:16',
          'name': 'World Cup',
          'sport': {
            'id': 'sr:sport:1',
            'name': 'Soccer',
          },
          'category': {
            'id': 'sr:category:4',
            'name': 'International',
          },
        },
        'competitors': [
          {
            'id': 'sr:competitor:4694',
            'name': 'Russia',
            'country': 'Russia',
            'country_code': 'RUS',
            'abbreviation': 'RUS',
            'qualifier': 'home',
          },
          {
            'id': 'sr:competitor:4834',
            'name': 'Saudi Arabia',
            'country': 'Saudi Arabia',
            'country_code': 'SAU',
            'abbreviation': 'KSA',
            'qualifier': 'away',
          },
        ],
        'venue': {
          'id': 'sr:venue:949',
          'name': 'Luzhniki Stadium',
          'capacity': 78011,
          'city_name': 'Moscow',
          'country_name': 'Russia',
          'map_coordinates': '55.715556,37.553611',
          'country_code': 'RUS',
        },
      },
      'sport_event_status': {
        'status': 'closed',
        'match_status': 'ended',
        'home_score': 0,
        'away_score': 5,
        'winner_id': 'sr:competitor:4694',
        'period_scores': [
          {
            'home_score': 0,
            'away_score': 0,
            'type': 'regular_period',
            'number': 1,
          },
          {
            'home_score': 0,
            'away_score': 5,
            'type': 'regular_period',
            'number': 2,
          },
        ],
      },
    },
    {
      'sport_event': {
        'id': 'sr:match:13246028',
        'scheduled': '2018-06-18T12:00:00+00:00',
        'start_time_tbd': false,
        'tournament_round': {
          'type': 'group',
          'number': 1,
          'group': 'F',
          'tournament_match_number': 12,
        },
        'season': {
          'id': 'sr:season:48238',
          'name': 'World Cup',
          'start_date': '2018-06-14',
          'end_date': '2018-07-16',
          'year': '2018',
          'tournament_id': 'sr:tournament:16',
        },
        'tournament': {
          'id': 'sr:tournament:16',
          'name': 'World Cup',
          'sport': {
            'id': 'sr:sport:1',
            'name': 'Soccer',
          },
          'category': {
            'id': 'sr:category:4',
            'name': 'International',
          },
        },
        'competitors': [
          {
            'id': 'sr:competitor:4688',
            'name': 'Sweden',
            'country': 'Sweden',
            'country_code': 'SWE',
            'abbreviation': 'SWE',
            'qualifier': 'home',
          },
          {
            'id': 'sr:competitor:4735',
            'name': 'Republic of Korea',
            'country': 'Republic of Korea',
            'country_code': 'KOR',
            'abbreviation': 'KOR',
            'qualifier': 'away',
          },
        ],
        'venue': {
          'id': 'sr:venue:26170',
          'name': 'Nizhny Novgorod Stadium',
          'capacity': 43319,
          'city_name': 'Nizhny Novgorod',
          'country_name': 'Russia',
          'map_coordinates': '56.337500,43.963333',
          'country_code': 'RUS',
        },
      },
      'sport_event_status': {
        'status': 'closed',
        'match_status': 'ended',
        'home_score': 1,
        'away_score': 0,
        'winner_id': 'sr:competitor:4688',
        'period_scores': [
          {
            'home_score': 0,
            'away_score': 0,
            'type': 'regular_period',
            'number': 1,
          },
          {
            'home_score': 1,
            'away_score': 0,
            'type': 'regular_period',
            'number': 2,
          },
        ],
      },
    },
    {
      'sport_event': {
        'id': 'sr:match:13245922',
        'scheduled': '2018-06-15T12:00:00+00:00',
        'start_time_tbd': false,
        'tournament_round': {
          'type': 'group',
          'number': 2,
          'group': 'A',
          'tournament_match_number': 2,
        },
        'season': {
          'id': 'sr:season:48238',
          'name': 'World Cup 2018',
          'start_date': '2018-06-14',
          'end_date': '2018-07-16',
          'year': '2018',
          'tournament_id': 'sr:tournament:16',
        },
        'tournament': {
          'id': 'sr:tournament:16',
          'name': 'World Cup',
          'sport': {
            'id': 'sr:sport:1',
            'name': 'Soccer',
          },
          'category': {
            'id': 'sr:category:4',
            'name': 'International',
          },
        },
        'competitors': [
          {
            'id': 'sr:competitor:4758',
            'name': 'Egypt',
            'country': 'Egypt',
            'country_code': 'EGY',
            'abbreviation': 'EGY',
            'qualifier': 'home',
          },
          {
            'id': 'sr:competitor:4725',
            'name': 'Uruguay',
            'country': 'Uruguay',
            'country_code': 'URY',
            'abbreviation': 'URU',
            'qualifier': 'away',
          },
        ],
        'venue': {
          'id': 'sr:venue:5206',
          'name': 'Ekaterinburg Arena',
          'capacity': 33061,
          'city_name': 'Ekaterinburg',
          'country_name': 'Russia',
          'map_coordinates': '56.832500,60.573611',
          'country_code': 'RUS',
        },
      },
      'sport_event_status': {
        'status': 'closed',
        'match_status': 'ended',
        'home_score': 0,
        'away_score': 1,
        'winner_id': 'sr:competitor:4725',
        'period_scores': [
          {
            'home_score': 0,
            'away_score': 0,
            'type': 'regular_period',
            'number': 1,
          },
          {
            'home_score': 0,
            'away_score': 1,
            'type': 'regular_period',
            'number': 2,
          },
        ],
      },
    },
  ],
};
