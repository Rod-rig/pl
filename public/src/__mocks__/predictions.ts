/* tslint:disable: object-literal-key-quotes object-literal-sort-keys*/
export const predictions = [
  {
    awayTeam: "Tottenham Hotspur",
    homeTeam: "Newcastle United",
    homeScore: 2,
    awayScore: 2,
    status: 0,
  },
  {
    awayTeam: "West Ham United",
    homeTeam: "Liverpool FC",
    homeScore: 1,
    awayScore: 2,
    status: 1,
  },
  {
    awayTeam: "Manchester City",
    homeTeam: "Wolverhampton Wanderers",
    homeScore: 6,
    awayScore: 1,
    status: -1,
  },
  {
    awayTeam: "Fulham FC",
    homeTeam: "Cardiff City",
    homeScore: 1,
    awayScore: 1,
    status: 0,
  },
];
