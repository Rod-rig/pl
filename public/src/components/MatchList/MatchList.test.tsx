import {mount} from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Loader, MatchList} from '../';
import {matchListMock} from '../../__mocks__';

describe('MatchList', () => {
  it('should render loader', () => {
    const notRenderedComp = mount(
      <MatchList
        store={{
          data: {
            results: undefined,
          },
          isLoaded: false,
          url: 'test',
        }}
      />);
    expect(notRenderedComp.find(Loader)).toHaveLength(1);
  });

  it('should render correctly', () => {
    const comp = (
      <MatchList
        store={{
          data: {
            results: matchListMock.results,
          },
          isLoaded: true,
          url: 'test',
        }}
      />);
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
