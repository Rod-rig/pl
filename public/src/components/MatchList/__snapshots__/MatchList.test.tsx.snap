// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchList should render correctly 1`] = `
<ul
  className="MuiList-root-3"
>
  <div
    className="class_1-title-2"
  >
    Round 
    1
  </div>
  <div
    className="MuiPaper-root-7 MuiPaper-elevation2-11 MuiPaper-rounded-8 class_1-round-1"
  >
    <div
      className="MuiButtonBase-root-50 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-divider-46 MuiListItem-button-48"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiListItemText-root-53 Component-text-39 Component-right-37"
      >
        <div>
          Russia
        </div>
      </div>
      <div
        className="Component-score-38"
      >
        <div>
          0
        </div>
        <div
          className="Component-dash-34"
        >
          -
        </div>
        <div>
          5
        </div>
      </div>
      <div
        className="MuiListItemText-root-53 Component-text-39"
      >
        <div>
          Saudi Arabia
        </div>
      </div>
      <span
        className="MuiTouchRipple-root-59"
      />
    </div>
    <div
      className="MuiButtonBase-root-50 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-divider-46 MuiListItem-button-48"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiListItemText-root-53 Component-text-39 Component-right-37"
      >
        <div>
          Sweden
        </div>
      </div>
      <div
        className="Component-score-38"
      >
        <div>
          1
        </div>
        <div
          className="Component-dash-34"
        >
          -
        </div>
        <div>
          0
        </div>
      </div>
      <div
        className="MuiListItemText-root-53 Component-text-39"
      >
        <div>
          Republic of Korea
        </div>
      </div>
      <span
        className="MuiTouchRipple-root-59"
      />
    </div>
  </div>
  <div
    className="class_1-title-2"
  >
    Round 
    2
  </div>
  <div
    className="MuiPaper-root-7 MuiPaper-elevation2-11 MuiPaper-rounded-8 class_1-round-1"
  >
    <div
      className="MuiButtonBase-root-50 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-divider-46 MuiListItem-button-48"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiListItemText-root-53 Component-text-39 Component-right-37"
      >
        <div>
          Egypt
        </div>
      </div>
      <div
        className="Component-score-38"
      >
        <div>
          0
        </div>
        <div
          className="Component-dash-34"
        >
          -
        </div>
        <div>
          1
        </div>
      </div>
      <div
        className="MuiListItemText-root-53 Component-text-39"
      >
        <div>
          Uruguay
        </div>
      </div>
      <span
        className="MuiTouchRipple-root-59"
      />
    </div>
  </div>
</ul>
`;
